app.rb

```diff
@@ -0,0 +1,94 @@
+# app.rb
+
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'nameable'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'decorator'
+require_relative 'capitalize_decorator'
+require_relative 'trimmer_decorator'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    puts 'List of Books:'
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    puts
+  end
+
+  def list_people
+    puts 'List of People:'
+    @people.each { |person| puts "ID: #{person.id}, Name: #{person.name}, Age: #{person.age}" }
+    puts
+  end
+
+  def create_person(name, age, type, specialization = nil)
+    if type == 'teacher'
+      person = Teacher.new(specialization, name, age)
+    elsif type == 'student'
+      person = Student.new(name, age)
+    else
+      puts 'Invalid person type.'
+      return
+    end
+
+    @people << person
+    puts "Person with ID #{person.id} was created."
+  end
+
+  def create_book(title, author)
+    book = Book.new(title, author)
+    @books << book
+    puts "Book '#{book.title}' by #{book.author} was created."
+  end
+
+  def create_rental
+    puts 'Enter the rental date (yyyy-mm-dd):'
+    date = gets.chomp
+    puts 'Enter the book title for the rental:'
+    title = gets.chomp
+
+    book = @books.find { |b| b.title == title }
+    if book.nil?
+      puts "Book with title '#{title}' not found."
+      return
+    end
+
+    puts 'Enter the person ID for the rental:'
+    person_id = gets.chomp.to_i
+
+    person = @people.find { |p| p.id == person_id }
+    if person.nil?
+      puts "Person with ID #{person_id} not found."
+      return
+    end
+
+    rental = Rental.new(date, book, person)
+    @rentals << rental
+
+    puts "Rental of '#{book.title}' by #{book.author} to #{person.correct_name} on #{rental.date} has been created."
+  end
```

Comment:
- [ ] The app crashes when I try to create a rental. Kindly inspect and fix that.

  ![image](https://github.com/rahimialijan/School-Library/assets/81468675/2716b7a0-4528-4452-9058-d9dfc66d7398)

Reply:
Thanks for you reviewing my PR, But it correctly works on my system. I will try my best and solve the issue